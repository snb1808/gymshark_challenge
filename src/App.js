import { Component } from 'react';
import './App.css';
import CardColumns from "react-bootstrap/CardColumns";
import { getData } from './adapters/API';
import ExerciseCard from "./components/ExerciseCard/ExerciseCard";

class App extends Component {
  state = { sex: 'female' };

  componentDidMount = () => {
      getData().then(({ exercises }) =>
  this.setState({ exercises }));
  };

  render = () => {
      const { exercises, sex } = this.state;

      return (
          <CardColumns>
              { exercises && exercises.map((exercise, index) =>
                 <ExerciseCard key={index} exercise={exercise} sex={sex}/>
              )}
          </CardColumns>
  )};
}

export default App;
