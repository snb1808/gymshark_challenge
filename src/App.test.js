import { shallow, mount } from 'enzyme';
import App from './App';

describe('App', () => {
  let componentDidMount;
  let getData;

  beforeEach(() => {
    componentDidMount = jest.fn();

    getData = jest.fn().mockResolvedValue({
        exercises: [{
          name: 'Bicep Curls',
          transcript: 'Bicep curls are bicep curls',
          male: { image: 'www.mensimage.com' },
          female: { image: 'www.womensimage.com' },
        },
          {
          name: 'Dumbell Squats',
          transcript: 'Dumbell squats are dumbell squatss',
          male: { image: 'www.mensimage.com' },
          female: { image: 'www.womensimage.com' },
        }]
    })
  });

  it('should render correctly without exercise cards ', () => {
    const component = shallow(<App />);

    expect(component.debug()).toMatchSnapshot();
  });

  it('should call the API to get exercise data', () => {
    const component = mount(<App />);

    expect(getData).toHaveBeenCalled();
  });
});