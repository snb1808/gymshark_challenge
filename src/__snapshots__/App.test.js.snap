// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly without exercise cards  1`] = `
"<div className=\\"app\\">
  <Header filterExercises={[Function (anonymous)]} setGender={[Function (anonymous)]} gender=\\"female\\" />
  <CardColumns />
</div>"
`;
